gspread/worksheet.py

```diff
@@ -1824,3 +1833,129 @@ def delete_named_range(self, named_range_id):
+        :param str dimension: The dimension to group, can be one of
+            ``ROWS`` or ``COLUMNS``.
+        """
+        body = {
+            "requests": [
+                {
+                    "addDimensionGroup": {
+                        "range": {
+                            "sheetId": self.id,
+                            "dimension": dimension,
+                            "startIndex": start,
+                            "endIndex": end,
+                        },
+                    }
+                }
+            ]
+        }
+
+        return self.spreadsheet.batch_update(body)
+
+    def group_columns(self, start, end):
+        """
+        Group columns in order to hide them in the UI.
+
+        .. note::
+
+            API behavior with nested groups and non matching ``[start:end[``
+            range can be found here
+
+            https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets/request#AddDimensionGroupRequest
+
+        :param int start: The start (inclusive) of the group
+        :param int end: The end (exclusive) of the group
+        """
+        return self._add_dimension_group(start, end, Dimension.cols)
+
+    def group_rows(self, start, end):
+        """
+        Group rows in order to hide them in the UI.
+
+        .. note::
+
+            API behavior with nested groups and non matching ``[start:end[``
+            range can be found here
+
+            https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets/request#AddDimensionGroupRequest
+
+        :param int start: The start (inclusive) of the group
+        :param int end: The end (exclusive) of the group
+        """
+        return self._add_dimension_group(start, end, Dimension.rows)
+
+    def _delete_dimension_group(self, start, end, dimension):
+        """delete a dimension group in this sheet"""
+        body = {
+            "requests": [
+                {
+                    "deleteDimensionGroup": {
+                        "range": {
+                            "sheetId": self.id,
+                            "dimension": dimension,
+                            "startIndex": start,
+                            "endIndex": end,
+                        }
+                    }
+                }
+            ]
+        }
+
+        return self.spreadsheet.batch_update(body)
+
+    def delete_grouped_columns(self, start, end):
+        """
+        Delete a grouped columns
+
+        .. note::
+
+            API behavior with nested groups and non matching ``[start:end[``
+            range can be found here
+
+            https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets/request#DeleteDimensionGroupRequest
+
+        :param int start: The start (inclusive) of the group
+        :param int end: The end (exclusive) of the group
+        """
+        return self._delete_dimension_group(start, end, Dimension.cols)
+
+    def delete_grouped_rows(self, start, end):
```

Comment:
@lavigne958 maybe this could be named this "delete_row_grouping"? If it's named "delete_grouped_rows" I would assume it deletes group **and the rows**, not just the group. 

Similar for columns.

Reply:
I will go for the `Dimension` term exactly like in the API that will it simpler for users to expect the right behavior.